bitcoin/Cargo.toml

```diff
@@ -26,15 +26,16 @@ bitcoinconsensus-std = ["bitcoinconsensus/std", "std"]
 # The no-std feature doesn't disable std - you need to turn off the std feature for that by disabling default.
 # Instead no-std enables additional features required for this crate to be usable without std.
 # As a result, both can be enabled without conflict.
-std = ["secp256k1/std", "bitcoin_hashes/std", "bech32/std", "bitcoin-internals/std"]
-no-std = ["core2", "bitcoin_hashes/alloc", "bitcoin_hashes/core2", "secp256k1/alloc"]
+std = ["secp256k1/std", "bitcoin_hashes/std", "bech32/std", "bitcoin-internals/std", "bitcoin-hex/std"]
+no-std = ["core2", "bitcoin_hashes/alloc", "bitcoin_hashes/core2", "secp256k1/alloc", "bitcoin-hex/alloc", "bitcoin-hex/core2"]
 
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 bitcoin-internals = { path = "../internals" }
+bitcoin-hex = { path = "../hex", default-features = false }
```

Comment:
Consider using `hex = { package = "bitcoin-hex" ... }` to make access easier. (Like I did in hashes with internals, was thinking about doing it above too.)

Also I guess sooner or later we will have internals depend on hex.

Reply:
I used `pub extern crate bitcoin_hex as hex;` in `bitcoin/src/lib.rs` - achieves the same goal, I couldn't work out what was different so I went with the extern because we already have it for `hashes`.

Reply:
Started a discussion on something I've been wondering about for a while that relates to this: https://github.com/rust-bitcoin/rust-bitcoin/discussions/1661